livekit-agents/livekit/agents/pipeline/pipeline_agent.py

```diff
@@ -754,83 +781,104 @@ def _commit_user_question_if_needed() -> None:
 
             new_function_calls = llm_stream.function_calls
 
-            for i in range(self._opts.max_nested_fnc_calls):
-                self.emit("function_calls_collected", new_function_calls)
+            self.emit("function_calls_collected", new_function_calls)
 
-                called_fncs = []
-                for fnc in new_function_calls:
-                    called_fnc = fnc.execute()
-                    called_fncs.append(called_fnc)
-                    logger.debug(
-                        "executing ai function",
+            called_fncs = []
+            for fnc in new_function_calls:
+                called_fnc = fnc.execute()
+                called_fncs.append(called_fnc)
+                logger.debug(
+                    "executing ai function",
+                    extra={
+                        "function": fnc.function_info.name,
+                        "speech_id": speech_handle.id,
+                    },
+                )
+                try:
+                    await called_fnc.task
+                except Exception as e:
+                    logger.exception(
+                        "error executing ai function",
                         extra={
                             "function": fnc.function_info.name,
                             "speech_id": speech_handle.id,
                         },
```

Comment:
Do we still allow nested function calls (For the same SpeechHandle). Seems like this changed quite a bit. Just making sure

Reply:
Yes, if there is any nested function calls they will be processed in the nested speech handle recursively.